### PR TITLE
feat: install TypeScript

### PR DESCRIPTION
**Pull request by charlesfries**
- Install TypeScript
- Install DefinitelyTyped packages from Ember addon blueprint
- Install  `@types/ember-data__model`
- Configure eslint for TS
- Type `AbilitiesService`

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -63,6 +66,29 @@
     "@embroider/test-setup": "^0.42.3",
     "@glimmer/component": "^1.0.4",
     "@glimmer/tracking": "^1.0.4",
+    "@types/ember__application": "^3.16.3",
+    "@types/ember__array": "^3.16.4",
+    "@types/ember__component": "^3.16.6",
+    "@types/ember__controller": "^3.16.5",
```
Comment (charlesfries) [inline package.json:72]: Probably can remove

File: package.json [hunk 2]
```diff
@@ -63,6 +66,29 @@
...
+    "@types/ember__debug": "^3.16.4",
+    "@types/ember__engine": "^3.16.3",
+    "@types/ember__error": "^3.16.1",
+    "@types/ember__object": "^3.12.6",
+    "@types/ember__polyfills": "^3.12.1",
+    "@types/ember__routing": "^3.16.15",
+    "@types/ember__runloop": "^3.16.3",
+    "@types/ember__service": "^3.16.1",
+    "@types/ember__string": "^3.16.3",
+    "@types/ember__template": "^3.16.1",
```
Comment (charlesfries) [inline package.json:82]: Probably can remove

File: package.json [hunk 3]
```diff
@@ -63,6 +66,29 @@
     "@embroider/test-setup": "^0.42.3",
     "@glimmer/component": "^1.0.4",
     "@glimmer/tracking": "^1.0.4",
+    "@types/ember__application": "^3.16.3",
+    "@types/ember__array": "^3.16.4",
+    "@types/ember__component": "^3.16.6",
```
Comment (charlesfries) [inline package.json:71]: Probably can remove

File: package.json [hunk 4]
```diff
@@ -63,6 +66,29 @@
...
+    "@types/ember__controller": "^3.16.5",
+    "@types/ember__debug": "^3.16.4",
+    "@types/ember__engine": "^3.16.3",
+    "@types/ember__error": "^3.16.1",
+    "@types/ember__object": "^3.12.6",
+    "@types/ember__polyfills": "^3.12.1",
+    "@types/ember__routing": "^3.16.15",
```
Comment (charlesfries) [inline package.json:78]: Probably can remove

**Comment by Exelord** [comment]
I know, need that as well :) Ember-can is awaiting a major upgrade soon.

**Comment by Exelord** [comment]
It's happening :) some changes just landed. TS will be welcome for upcoming v5

**Comment by guidojw** [comment]
@Exelord Is it still happening? 😋  
I am slowly but surely converting our Ember app to TS and this addon is one of the few that doesn't come with types yet.

**Comment by Exelord** [comment]
Unfortunately I don't have enough time to work on it anymore :/ so the progres on next upgrade is on hold.

**Comment by esbanarango** [comment]
@Exelord Is there anything we could help with to either merge this PR or move the V5 forward? 🙏

**Comment by Exelord** [comment]
Hey @esbanarango,

As mentioned, currently I do not have time to maintain this package. I am really sorry for that. I would recommend maintaining own for with changes needed for your project taking into account how small this project is.

**Comment by acorncom** [comment]
@Exelord are you able to hand over maintenance to others who do have the time? Completely understood that you have other time constraints on your end, yet there's still interest in having a shared add-on that the community uses for this ...

**Comment by Exelord** [comment]
Hi @acorncom,

I have the rights to do it, but I am not the owner of this addon.

@rlivsey would you like to choose a new maintainer or pass ownership to someone?

**Comment by RobbieTheWagner** [comment]
@Exelord @rlivsey might I suggest transferring this to adopted-ember-addons?

**Comment by mkszepp** [comment]
@rlivsey would be nice to transfer this addon to adopted-ember-addons or to get a new owner, so that the addon can be migrate to TS and addon v2

**Comment by Exelord** [comment]
I cannot move the repo, but as the author is not responding, I will grant maintenance permission. 

Who would like to take it over?

**Comment by esbanarango** [comment]
@Exelord I can help

**Comment by RobbieTheWagner** [comment]
I'm also willing to help. Getting this in would be great 😄

**Comment by Exelord** [comment]
Done. Please sync with each other on next steps :) @RobbieTheWagner @esbanarango

**Comment by RobbieTheWagner** [comment]
Thanks @Exelord! @esbanarango do you have the cycles to take a look at getting this ready to merge? I won't have availability for probably a few weeks.

**Comment by esbanarango** [comment]
@RobbieTheWagner Yeah, I'll take a look! @charlesfries Let me know once this is ready 🤝

**Comment by charlesfries** [comment]
@esbanarango should be good now

File: package.json [hunk 5]
```diff
@@ -87,7 +119,9 @@
     "prettier": "^2.5.0",
     "qunit": "^2.17.2",
     "qunit-dom": "^2.0.0",
-    "sinon": "^12.0.1",
+    "rimraf": "^5.0.5",
+    "sinon": "^17.0.1",
```
Comment (charlesfries) [inline package.json:123]: Needed to update this to solve a build error:

```
ERROR in ./node_modules/@sinonjs/fake-timers/src/fake-timers-src.js 104:479-504
Module not found: Error: Can't resolve 'util' in 'C:\Users\charlesfries\Repositories\ember-can\node_modules\@sinonjs\fake-timers\src'```

File: package.json [hunk 6]
```diff
@@ -54,11 +57,40 @@
   },
   "devDependencies": {
     "@ember/optional-features": "^2.0.0",
-    "@ember/test-helpers": "^2.6.0",
+    "@ember/test-helpers": "^3.3.0",
```
Comment (charlesfries) [inline package.json:60]: Needed to update this to fix a Glint TS error:

```
node_modules/@glint/environment-ember-loose/-private/dsl/integration-declarations.d.ts:107:16 - error TS2665: Invalid module name in augmentation. Module '@ember/test-helpers' resolves to an untyped module at 'C:/Users/xxx/ember-can/node_modules/@ember/test-helpers/index.js', which cannot be augmented.
```

**Pull request closed without merging**